sdks/python/setup.py

```diff
@@ -125,6 +125,7 @@ def get_version():
     'pyvcf>=0.6.8,<0.7.0; python_version < "3.0"',
     'pyyaml>=3.12,<4.0.0',
     'typing>=3.6.0,<3.7.0; python_version < "3.5.0"',
+    'mmh3>=2.5.1; python_version >= "2.7"',
```

Comment:
A few questions:
- What is this dependency? The pypi page says this is a wrapper. Does it require other things to be installed? Is this the only option we have?
- You do need the python_version >= "2.7" part, because all our sdks only support py >= 2.7
- Can you add an upper bound here. Maybe <3.0.0

Reply:
Ahmet, you are correct, we need this package only for python 2.7. With python 3, buildin hash function is different than the one from 2.7 and it works well. For 2.7, we still need to use mmh3. mmh3 is a wrapper of MurmurHash3, however, MurmurHash3 only supports from python 3.2. I don't think mmh3 requires other things to be installed. Please let me know if we want to investigate more.

Reply:
Could you clarify why the builtin hash function does not work well in 2.7?

Reply:
Thanks for asking this question. I was able to dig into more and found an unexpected reason.
From 3.4, Python used SpiHash algorithm, before that, it used FNV algorithm. Though SpiHash reduced collision, I don't think it's noticeable from my test. So I saw how hash values are distributed. Graphs at https://docs.google.com/spreadsheets/d/1QpNAorUhUY1Nq3b4QRMfbB9Rupq3ZC9FshunIyRRfH0/edit#gid=0 shows how hash values are distributed with different hash algorithm given exact the same data set. Builtin hash function with Python 2 is not as uniformly distributed as other hash functions (builtin hash function with Python 3 and mmh3). Though it's not obvious from the graphs that hash values from mmh3 algorithms are more uniformed distributed than the ones from builtin hash function with python 3, test results shows that mmh3 has more uniformly distributed hash values.

Why does it matter to approximate unique count? We are calculating sample space with hash values and calculate population space from the sample space. More uniformly distributed hash values give us more accurate sample space thus we can calculate more accurate population space. The difference is more obvious when we have more duplicate records with input.

A test case is generating 10000 elements within range of [0, 1000]. This test consistently fails if we use builtin hash function with py2.7, and fails sometimes with builtin hash function with py3 and hasn't fail with mmh3.

Reply:
![mmh3](https://user-images.githubusercontent.com/16039146/57813427-e66de300-7724-11e9-90d0-432b465aedad.png)
![py2](https://user-images.githubusercontent.com/16039146/57813429-e66de300-7724-11e9-8d6a-c7577dcb540a.png)
![py3](https://user-images.githubusercontent.com/16039146/57813430-e66de300-7724-11e9-80d1-7220cf60af8a.png)